src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,512 @@
+    : QWidget(nullptr),
+      annDnaView(_annDnaView),
+      savableWidget(this, GObjectViewUtils::findViewByName(annDnaView->getName())) {
+    setupUi(this);
+    parametersMinMaxSpinBoxes = { { sbMinRequireGibbs, sbMaxRequireGibbs },
+                                  { spMinRequireMeltingTeml, spMaxRequireMeltingTeml },
+                                  { spMinRequireOverlapLength, spMaxRequireOverlapLength },
+                                  { sbLeftAreaStart, sbLeftAreaEnd },
+                                  { sbRightAreaStart, sbRightAreaEnd } };
+
+    userPrimersShowHideGroup->init(USER_PRIMERS_SHOW_HIDE_ID, tr("User primers"), wgtUserPrimers, true);
+    generateSequenceShowHideGroup->init(GENERATE_SEQUENCE_SHOW_HIDE_ID,
+                                        tr("Choose generated sequences as user primer's end"),
+                                        wgtGenerateSequence,
+                                        true);
+    parametersOfPrimingSequencesShowHideGroup->init(PARAMETERS_OF_PRIMING_SEQUENCES_SHOW_HIDE_ID,
+                                                    tr("Parameters of priming sequences"),
+                                                    wgtParametersOfPrimingSequences,
+                                                    true);
+    parameters2ExcludeInWholePrimersShowHideGroup->init(PARAMETERS_2_EXCLUDE_IN_WHOLE_PRIMERS_SHOW_HIDE_ID,
+                                                        tr("Parameters to exclude in whole primers"),
+                                                        wgtParameters2ExcludeInWholePrimers,
+                                                        true);
+    areasForPrimingShowHideGroup->init(SELECT_AREAS_FOR_PRIMING_SHOW_HIDE_ID,
+                                       tr("Select areas for priming search"),
+                                       wgtAreasForPriming,
+                                       true);
+    openBackboneSequenceShowHideGroup->init(OPEN_BACKBONE_SEQUENCE_SHOW_HIDE_ID,
+                                            tr("Open the backbone sequence"),
+                                            wgtOpenBackboneSequence,
+                                            true);
+    otherSequencesInPCRShowHideGroup->init(OTHER_SEQUENCES_IN_PCR_REACTION_SHOW_HIDE_ID,
+                                           tr("Other sequences in PCR reaction"),
+                                           wgtOtherSequencesInPcr,
+                                           true);
+
+    auto seqLength = annDnaView->getActiveSequenceContext()->getSequenceLength();
+
+    // Default values, have been provided by the customer
+    // Left area: from "1" to "10% of sequence length"
+    // Right area: from "50% of sequence length" to "60% of sequence length"
+    sbLeftAreaStart->setValue(1);
+    sbLeftAreaEnd->setValue(seqLength / 10);
+    sbRightAreaStart->setValue(seqLength / 10 * 5);
+    sbRightAreaEnd->setValue(seqLength / 10 * 6);
```

Comment:
Loss of accuracy, or did the customer want to have starting coordinates that are multiples of five? Use `static_cast<double>(seqLength)` or `seqLength * 0.5`.